main.tf

```diff
@@ -7,6 +7,7 @@ locals {
   availability_zone    = "${var.availability_zone != "" ? var.availability_zone : data.aws_subnet.default.availability_zone}"
   ami                  = "${var.ami != "" ? var.ami : data.aws_ami.default.image_id}"
   root_volume_type     = "${var.root_volume_type != "" ? var.root_volume_type : data.aws_ami.info.root_device_type}"
+  user_data            = "${length(var.custom_user_data) > 0 ? join("", var.custom_user_data) : join("", data.template_file.user_data.*.rendered)}"
```

Comment:
Should we use `var.custom_user_data` instead of `join("", var.custom_user_data) `?

Reply:
in this case will be error during plan\apply, because trueval\falseeval should be string